### PR TITLE
[RLlib] PyTorch version of ARS (Augmented Random Search).

### PR DESCRIPTION
**Pull request by sven1977**
NOTE: Please merge https://github.com/ray-project/ray/pull/8104 first (PyTorch ES).

This PR implements a PyTorch version of RLlib's ARS algorithm using RLlib's functional algo builder API. It also adds a regression test for ARS (torch) on CartPole.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/es/es_torch_policy.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+        for k in sorted(theta_dict.keys()):
+            shape = policy.param_shapes[k]
+            num_params = int(np.prod(shape))
+            new_theta_dict[k] = torch.from_numpy(
+                np.reshape(theta[pos:pos + num_params], shape))
+            pos += num_params
+        policy.model.load_state_dict(new_theta_dict)
+
+    def _get_flat_weights(policy):
```
Comment (ericl) [inline rllib/agents/es/es_torch_policy.py:40]: Should these be common methods of TorchPolicy?

**Review by ericl — APPROVED** [review]
LGTM, same question for benchmarks as the ES pr.

File: doc/source/rllib-algorithms.rst [hunk 2]
```diff
@@ -405,7 +405,7 @@ Derivative-free
 
 Augmented Random Search (ARS)
 -----------------------------
-|tensorflow|
+|pytorch| |tensorflow|
```
Comment (ericl) [inline doc/source/rllib-algorithms.rst:408]: Don't forget to update the sigils in rllib-toc.rst

Comment (sven1977) [reply doc/source/rllib-algorithms.rst:408]: +1 Added to toc as well.

**Comment by sven1977** [comment]
Waiting for re-test, just in case...

**Pull request merged**